src/main/java/io/cdap/plugin/google/sheets/source/GoogleSheetsSourceConfig.java

```diff
@@ -0,0 +1,872 @@
+        return Schema.of(Schema.Type.STRING);
+      }
+    } else if (value.getBoolValue() != null) {
+      return Schema.of(Schema.Type.BOOLEAN);
+    } else {
+      return Schema.of(Schema.Type.STRING);
+    }
+  }
+
+  private LinkedHashMap<Integer, ColumnComplexSchemaInfo> defaultGeneratedHeaders(int length) {
+    LinkedHashMap<Integer, ColumnComplexSchemaInfo> headers = new LinkedHashMap<>();
+    for (int i = 1; i <= length; i++) {
+      headers.put(i - 1, new ColumnComplexSchemaInfo(ColumnAddressConverter.getColumnName(i),
+        Schema.of(Schema.Type.STRING)));
+    }
+    return headers;
+  }
+
+  public void validateMetadata(FailureCollector collector) {
+    if (!containsMacro(EXTRACT_METADATA) && !containsMacro(METADATA_CELLS)
+      && !containsMacro(FIRST_HEADER_ROW) && !containsMacro(LAST_HEADER_ROW)
+      && !containsMacro(FIRST_FOOTER_ROW) && !containsMacro(LAST_FOOTER_ROW)
+      && extractMetadata
+        && checkPropertyIsSet(collector, metadataCells, METADATA_CELLS, METADATA_CELLS_LABEL)
+        & checkPropertyIsValid(collector, firstFooterRow != null && getFirstHeaderRow() > -2,
+        FIRST_HEADER_ROW, getFirstHeaderRow(), FIRST_HEADER_ROW_LABEL)
+        & checkPropertyIsValid(collector, lastHeaderRow != null && getLastHeaderRow() > -2,
+        LAST_HEADER_ROW, getFirstHeaderRow(), LAST_HEADER_ROW_LABEL)
+        & checkPropertyIsValid(collector, firstFooterRow != null && getFirstFooterRow() > -2,
+        FIRST_FOOTER_ROW, firstFooterRow, FIRST_FOOTER_ROW_LABEL)
+        & checkPropertyIsValid(collector, lastFooterRow != null && getLastFooterRow() > -2,
+        LAST_FOOTER_ROW, lastFooterRow, LAST_FOOTER_ROW_LABEL)) {
+
+      if (getFirstHeaderRow() == -1 && getLastHeaderRow() == -1
+          && getFirstFooterRow() == -1 && getLastFooterRow() == -1) {
+        collector.addFailure("No header or footer rows specified.", null)
+            .withConfigProperty(firstHeaderRow)
+            .withConfigProperty(lastHeaderRow)
+            .withConfigProperty(firstFooterRow)
+            .withConfigProperty(lastFooterRow);
+        return;
+      }
+      if ((getFirstHeaderRow() != -1 && getLastHeaderRow() == -1)
+          || (getFirstHeaderRow() == -1 && getLastHeaderRow() != -1)) {
+        collector.addFailure("Both first and last rows should be specified or not for header.", null)
+            .withConfigProperty(firstHeaderRow)
+            .withConfigProperty(lastHeaderRow);
+        return;
+      }
+      if ((getFirstFooterRow() != -1 && getLastFooterRow() == -1)
+          || (getFirstFooterRow() == -1 && getLastFooterRow() != -1)) {
+        collector.addFailure("Both first and last rows should be specified or not for footer.", null)
+            .withConfigProperty(firstFooterRow)
+            .withConfigProperty(lastFooterRow);
+        return;
+      }
+      if (getFirstHeaderRow() > getLastHeaderRow()) {
+        collector.addFailure("Header first row cannot be less of header last row.", null)
```

Comment:
First row header index cannot be less than last row header index.

Reply:
done